harvard-lts/etd_dash_service
# Etd 185

**Add logging.**
* * *

**JIRA Ticket**: [(link)](https://jira.huit.harvard.edu/browse/ETD-185)

# What does this Pull Request do?
This pr adds logging, using the python "logging" module. It follows our logging standards:

https://wiki.harvard.edu/confluence/pages/viewpage.action?spaceKey=LibraryTechServices&title=SysDev+-+Logging+standards

It does both file system and console logging, and can be configured to log only to the console (as per standards, and as bridge to eventual centralized logging).

# How should this be tested?

- Check out this branch
- copy .env.example to .env and use dev queue creds
- start docker (change DockerfileLocal to use etd_submission_ready_<YOURINITIALS>
- look for ./logs/etd/etd_dash.log (note this maps to /logs/etd/etd_dash/etd_dash.log on dev, qa, prod)
- drop message on dev queue etd_submission_ready_<YOURINITIALS> as shown on README (Manually placing a message on the queue)
- look for hello world message in log
- also look for hello world in console: docker logs <localcontainerid>

On DEV:
- deploy to dev (done)
- Find container currently running for dash: docker stack ps ETD | grep Run|grep dash
- Tail log on container where dash is running: tail -f /logs/etd/etd_dash/etd_dash.log
- drop message on etd_submission_ready dev queue (again, see README)
- look for hello world message
- you can also see it in docker log (currently where console logging goes): docker service logs -f ETD-etd_dash

To test console only (locally):
- uncomment: #CONSOLE_LOGGING_ONLY=true  (note presence of this var works regardless of true/false, please confirm ok, I believe we do this in other apps
- see message in console but not filesys

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests?
- integration tests?

# Interested parties
@ives1227 , @cgoines, @awoods
